### PR TITLE
Convert project to esm

### PR DESCRIPTION
**Pull request by edenhermelin**
Close https://github.com/IonicaBizau/parse-url/issues/31 (perhaps also close https://github.com/IonicaBizau/parse-url/issues/32 )

This PR updates the project to use ESM and allows to import the new version of the `normalize-url` library

### CODE REVIEW

**Comment by edenhermelin** [comment]
Hey @IonicaBizau just checking in, if there's anything else to add to the PR so it'll be merged please let me know 🙏

**Comment by IonicaBizau** [comment]
@edenhermelin Sorry for the late reply on this — would this be backwards compatible with `require`?

Thank you very much!

**Comment by omry-hay** [comment]
Hi,
Any updates on this one?

The main issue we have is that the `normalize-url` version breaks our application in Safari due to [this bug](https://github.com/sindresorhus/normalize-url/issues/142) that was merged on version `7.0.1` of `normalize-url`.

Thanks

File: package.json [hunk 1]
```diff
@@ -32,6 +33,7 @@
   },
   "dependencies": {
     "is-ssh": "^1.3.0",
+    "normalize-url": "^7.0.3",
     "normalize-url": "^6.1.0",
```
Comment (talkor) [inline package.json:37]:
```suggestion
    "normalize-url": "^7.0.3",
```

Comment (edenhermelin) [reply package.json]: Thanks!

**Review by talkor — APPROVED** [review]

**Comment by edenhermelin** [comment]
> @edenhermelin Sorry for the late reply on this — would this be backwards compatible with `require`?
> 
> Thank you very much!

Hey @IonicaBizau , I think that might be OK but it depends on the configuration of the project that uses this library. Anyway, the PR allows to update the nested dependency (that fixes the usage for Safari), so I guess that it will be effective to add it as currently there are a lot of people blocked from using the library 😃

**Review by RLRabinowitz — APPROVED** [review]

**Comment by privatenumber** [comment]
FYI this is a breaking change and warrants a major version bump as CommonJS packages will no longer be able to `require()` it.

File: lib/index.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 "use strict"
```
Comment (privatenumber) [inline lib/index.js:1]: ESM files are strict by default

**Pull request merged**